### PR TITLE
Update lightgbm docs

### LINKED ISSUE #7502: [DOC-FIX] Populate API and usage examples for LightGBM module
### Willingness to contribute

No. I cannot contribute a documentation fix at this time.

### URL(s) with the issue

https://www.mlflow.org/docs/latest/python_api/mlflow.lightgbm.html

### Description of proposal (what needs changing)

Provide API examples for each public-facing API within `mlflow.lightgbm.py`

https://github.com/mlflow/mlflow/blob/50a0a869a58e63d064bf63744a782b034544b638/mlflow/lightgbm.py#L7-L9
https://github.com/mlflow/mlflow/blob/50a0a869a58e63d064bf63744a782b034544b638/mlflow/lightgbm.py#L81-L83
https://github.com/mlflow/mlflow/blob/50a0a869a58e63d064bf63744a782b034544b638/mlflow/lightgbm.py#L92-L95
https://github.com/mlflow/mlflow/blob/50a0a869a58e63d064bf63744a782b034544b638/mlflow/lightgbm.py#L111-L143
https://github.com/mlflow/mlflow/blob/50a0a869a58e63d064bf63744a782b034544b638/mlflow/lightgbm.py#L248-L288
https://github.com/mlflow/mlflow/blob/50a0a869a58e63d064bf63744a782b034544b638/mlflow/lightgbm.py#L343-L362
https://github.com/mlflow/mlflow/blob/50a0a869a58e63d064bf63744a782b034544b638/mlflow/lightgbm.py#L414-L458

To add relevant API usage examples in the same vein as the examples in:
https://github.com/mlflow/mlflow/blob/50a0a869a58e63d064bf63744a782b034544b638/mlflow/pytorch/__init__.py#L232-L303

NOTE: please use `..test-code-block:: python` for all code examples to ensure that unit tests on encapsulated code are tested.

NOTE: In order to perform local testing of code within comments:

``` shell
cd docs
make rsthtml  # should dump code blocks in `docs/.examples` as python scripts.
make test-examples
```

### PR DESCRIPTION
**Pull request by canerturkseven**
## Related Issues/PRs

Resolve #7502 

## What changes are proposed in this pull request?

Included code examples for public-facing lightgbm API's (save_model, log_model, load_model, autolog).

## How is this patch tested?

- [ ] Existing unit/integration tests
- [ ] New unit/integration tests
- [x] Manual tests (describe details, including test results, below)

## Does this PR change the documentation?

- [ ] No. You can skip the rest of this section.
- [x] Yes. Make sure the changed pages / sections render correctly in the documentation preview.

### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [x] Yes. Give a description of this change to be included in the release notes for MLflow users.

Added code examples for public-facing lightgbm API's.

### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [x] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [x] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: mlflow/lightgbm.py [hunk 1]
```diff
@@ -297,6 +346,52 @@ def log_model(
...
+        from lightgbm import LGBMRegressor
+        import numpy as np
+
+
+        def gen_data():
+            # Dummy data for f(x) = 2x
+            # X training data, y labels
+            X = np.arange(1, 51)
+            y = 2 * X
+            return X.reshape(-1, 1), y
```
Comment (harupy) [inline mlflow/lightgbm.py:363]: Can we use the iris dataset to shorten the example code?

Comment (canerturkseven) [reply mlflow/lightgbm.py:363]: Sure, I will make the necessary changes.

Comment (harupy) [reply mlflow/lightgbm.py:363]: @canerturkseven Can you also fix the DCO check failure!

Comment (canerturkseven) [reply mlflow/lightgbm.py]: Hi @harupy, I made the proposed changes and fixed the DCO failure.

**Review by harupy — APPROVED** [review]
LGTM!

**Pull request merged**